### PR TITLE
Implement episode and season search templates

### PR DESCRIPTION
**Pull request by p0psicles**
Implemented basic parsing of the templates for searching providers.

- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

This will allow users to create their own custom (templated) search string per show.
For example sport shows like ufc and formula 1 should become available.

~~For now a limitation is that you can only create 1 templated episode search string and 1 templated season search string.~~

Should implement #3708 

## Todo:
- [x] Move scene_exceptions table to main.db
- [x] Refactor scene_exceptions to return a named_tuple in stead of tuple. This will include all information for one exception row. Like series_id, indexer, season, episode_template, season_template.
- [x] Make generic_provider create episode_search_strings using the scene_exception episode_search_template.
- [x] Make generic_provider create season_search_strings using the scene_exception season_search_template.
- [x] Create UI component for editing season scene exceptions
- [x] Create UI component for creating the episode and season search templates. These will write to scene_exceptions table
- [x] Add template variable for `episode year` (%EY)
- [x] Add template variable for `month as a full name` (%b) 

### for vue, working on the dark template.
- [x] Don't forget to sync up

## Examples:
| Series Name                                | Example Result                                                                   | Search template         | Search String                                        | Example provider |
|--------------------------------------------|----------------------------------------------------------------------------------|-------------------------|------------------------------------------------------|------------------|
| Buried: Knights Templar And The Holy Grail | Buried.Knights.Templar.and.the.Holy.Grail.2018.Part1.720p.WEB.h264-CookieMonster | %SN.%Y.Part%E           | Buried Knights Templar And The Holy Grail.2018.Part1 | TD               |
| Burning Bush                               | Burning.Bush.E01.1080p.BluRay.x264-TAXES                                         | %SN.%0E                 | Burning Bush E01                                     |                  |
| Formula 1                                  | Formula1.2017.Abu.Dhabi.Grand.Prix.UNCUT.1080p.HDTV.x264-GRiP                    | %SN.%EY.%E.N       | Formula1.2017.Abu.Dhabi.Grand.Prix                   |                  |
| The Jeremy Kyle Show                       | ?                                                                                | %SN (%d %b %Y) | The Jeremy Kyle Show (8 February 2018)               |                  |

### CODE REVIEW

File: themes-default/slim/views/config.mako [hunk 1]
```diff
@@ -3,34 +3,34 @@
...
     <table class="infoTable" cellspacing="1" border="0" cellpadding="0" width="100%">
-        <tr v-if="release">
+        <tr v-if="config.release">
             <td><i class="icon16-config-application"></i> Medusa Info:</td>
             <td>
-                Branch: <a v-bind:href="sourceUrl + '/tree/' + branch" v-on:click="anonRedirect">{{branch}}</a><br>
-                Commit: <a v-bind:href="sourceUrl + '/commit/' + commitHash" v-on:click="anonRedirect">{{commitHash}}</a><br>
-                Version: <a v-bind:href="sourceUrl + '/releases/tag/' + release" v-on:click="anonRedirect">{{release}}</a><br>
-                Database: {{databaseVersion.major}}.{{databaseVersion.minor}}
+                Branch: <a v-bind:href="config.sourceUrl + '/tree/' + config.branch" v-on:click="anonRedirect">{{config.branch}}</a><br>
```
Comment (OmgImAlexis) [inline themes-default/slim/views/config.mako]: All of these `v-bind` can be removed and the `v-on` can be changed to `@`.

File: themes-default/slim/views/editShow.mako [hunk 2]
```diff
@@ -28,9 +22,9 @@
 % else:
     <h1 class="title">${title}</h1>
 % endif
-<div id="config" ${"class=\"summaryFanArt\"" if app.FANART_BACKGROUND else ""}>
-    <div id="config-content">
-        <form action="home/editShow" method="post">
+<div id="config-content">
+    <div id="config" v-bind:class="{ summaryFanArt: config.fanartBackground }">
+        <form v-on:submit.prevent="storeConfig">
```
Comment (OmgImAlexis) [inline themes-default/slim/views/editShow.mako]: Same goes for here with the `v-bind` and `v-on`.

File: themes/dark/templates/editShow.mako [hunk 3]
```diff
@@ -263,19 +308,41 @@ var startVue = function() {
...
                     }
                 }).then(response => {
                     log.info(response);
                 }).catch(err => {
                     log.error(err);
                 });
+            },
+            onChangeIgnoredWords: function(items) {
+		        console.log('Event from child component emitted', items);
+                this.seriesObj.config.release.ignoredWords = items.map(item => {return item.value});
```
Comment (OmgImAlexis) [inline themes/dark/templates/editShow.mako]: Change to `this.seriesObj.config.release.ignoredWords = items.map(item => item.value);`.

Comment (OmgImAlexis) [reply themes/dark/templates/editShow.mako]: Or even `this.seriesObj.config.release.ignoredWords = items.map(({value}) => value);``

File: themes/dark/templates/editShow.mako [hunk 4]
```diff
@@ -313,31 +309,29 @@ var startVue = function() {
...
+                const {validLanguages} = this.config.indexers.config.main;
+                if (validLanguages){
+                    return validLanguages.join(',');
                 }
+            },
+            ignoredWords() {
+                return this.transformToIndexedObject(this.series.config.release.ignoredWords);
+            },
+            requiredWords() {
+                return this.transformToIndexedObject(this.series.config.release.requiredWords);
```
Comment (OmgImAlexis) [inline themes/dark/templates/editShow.mako]: @p0psicles Instead of mutating the value in the DOM you should be doing in a computed property.

**Comment by p0psicles** [comment]
PR is becoming to big. I'm going to move the vue changes to editShow to a dedicated PR.

**Comment by p0psicles** [comment]
#3708

**Comment by p0psicles** [comment]
@firebladenoot i think this is what your interested in.

**Comment by p0psicles** [comment]
If the veuify postprocessing pr is merged, il continue on this.

@sharkykh mind reviewing the choosen solution?

Ive choosen to extend the scene_exceptions table. As then your allowed to create one searchtemplate per scene exception.

Thoughts?

**Comment by p0psicles** [comment]
https://github.com/pymedusa/Medusa/discussions/8953

File: medusa/name_parser/parser.py [hunk 5]
```diff
@@ -279,6 +279,12 @@ def _parse_series(result):
         new_absolute_numbers = []
 
         ex_season = scene_exceptions.get_season_from_name(result.series, result.series_name) or result.season_number
+        if ex_season == None:
+            ex_season = 1
+            log.info(
```
Comment (medariox) [inline medusa/name_parser/parser.py:284]: Won't this break support for most specials?

Comment (p0psicles) [reply medusa/name_parser/parser.py:284]: Didn't think of that. I'll check it out

**Pull request merged**